rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/system_interface.kt

```diff
@@ -118,11 +120,19 @@ class SimpleSystemInterface(var loggingConfiguration: LoggingConfiguration? = nu
     }
 
     override fun findProgram(name: String): Program? {
+        programFinders.forEach {
```

Comment:
here I think we could use `programFinders.mapNotNull { it.findProgram(name) }.firstOrNull()`

Reply:
In order to avoid useless calls to findRpgProgram, should I convert the collection to a sequence first?
`        programFinders.asSequence().mapNotNull {
            it.findRpgProgram(name)
        }.firstOrNull()
`